packages/smooth_app/lib/pages/product/nutrition_page_loaded.dart

```diff
@@ -74,41 +72,76 @@ class _NutritionPageLoadedState extends State<NutritionPageLoaded> {
 
   @override
   Widget build(BuildContext context) {
-    final AppLocalizations localizations = AppLocalizations.of(context);
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
     final LocalDatabase localDatabase = context.read<LocalDatabase>();
     final List<Widget> children = <Widget>[];
-    children.add(_switchNoNutrition(localizations));
-    if (!_unspecified) {
-      children.add(_getServingField(localizations));
-      children.add(_getServingSwitch(localizations));
+    children.add(_switchNoNutrition(appLocalizations));
+    if (!_noNutritionData) {
+      children.add(_getServingField(appLocalizations));
+      children.add(_getServingSwitch(appLocalizations));
       for (final OrderedNutrient orderedNutrient
           in _nutritionContainer.getDisplayableNutrients()) {
         children.add(
-          _getNutrientRow(localizations, orderedNutrient),
+          _getNutrientRow(appLocalizations, orderedNutrient),
         );
       }
-      children.add(_addNutrientButton(localizations));
+      children.add(_addNutrientButton(appLocalizations));
     }
 
     return WillPopScope(
-      //return a boolean to decide whether to return to previous page or not
-      onWillPop: () => _showCancelPopup(localizations),
+      onWillPop: () async {
+        if (!_isEdited()) {
+          return true;
+        }
+        final bool? pleaseSave = await showDialog<bool>(
+          context: context,
+          builder: (final BuildContext context) => SmoothAlertDialog(
+            close: true,
+            body:
+                Text(appLocalizations.edit_product_form_item_exit_confirmation),
+            title: appLocalizations.nutrition_page_title,
+            negativeAction: SmoothActionButton(
+              text: appLocalizations.ignore,
+              onPressed: () => Navigator.pop(context, false),
+            ),
+            positiveAction: SmoothActionButton(
+              text: appLocalizations.save,
+              onPressed: () => Navigator.pop(context, true),
+            ),
+          ),
+        );
+        if (pleaseSave == null) {
+          return false;
+        }
+        if (pleaseSave == false) {
+          return true;
+        }
```

Comment:
Shouldn't it return false here
Also then those 3 if's can be combined

Reply:
No, that's a bit confusing but now the dialog says: "You're about to leave the page. Do you want to save your changes?" Possible answers: "Ignore => false", "Save => true", close cross or dialog dismiss => null.
If the user answers "ignore" we just leave the page without saving, therefore we leave immediately (`return true` for `onWillPop`).

Therefore there are no if's to combine; besides, as a reader I prefer to read each step separately and flutter didn't complain about redundant if's. And most of the time when I try to code an if with negation/and/or combined I save a couple of lines of code but I waste time recomputing the if again and again when I re-read it :)